### PR TITLE
show warning message when ACR don't have image

### PR DESCRIPTION
**Pull request by SLdragon**
Fix issue : https://github.com/Microsoft/vscode-azure-iot-edge/issues/410

### CODE REVIEW

File: src/container/acrManager.ts [hunk 1]
```diff
@@ -121,6 +125,12 @@ export class AcrManager {
 
             const repoItems: vscode.QuickPickItem[] = [];
             const repos = JSON.parse(catalogResponse).repositories;
+
+            if (!repos) {
+                vscode.window.showWarningMessage("There is no repository in the registry.");
+                return;
```
Comment (adashen) [inline src/container/acrManager.ts]: return null is better?

File: src/container/acrManager.ts [hunk 2]
```diff
@@ -103,7 +103,12 @@ export class AcrManager {
     }
 
     private async selectAcrRepo(registryUrl: string, session: AzureSession): Promise<vscode.QuickPickItem> {
-        const acrRepoItem: vscode.QuickPickItem = await vscode.window.showQuickPick(this.loadAcrRepoItems(registryUrl, session), { placeHolder: "Select Repository", ignoreFocusOut: true });
+        const repoItems = await this.loadAcrRepoItems(registryUrl, session);
+        if (!repoItems) {
+            vscode.window.showWarningMessage("There is no repository in the registry.");
+            return;
+        }
+        const acrRepoItem: vscode.QuickPickItem = await vscode.window.showQuickPick(repoItems, { placeHolder: "Select Repository", ignoreFocusOut: true });
```
Comment (LazarusX) [inline src/container/acrManager.ts]: FYI, by updating the parameter passed to `showQuickPick` from a `Promise` to an `array`, VS Code will not show loading animations.

Comment (adashen) [reply src/container/acrManager.ts]: Good catch. @SLdragon, how about keep "promise", and return empty list of quickpickitem when there is no repository

**Comment by SLdragon** [comment]
Updated
empty-repository

File: src/container/acrManager.ts (hunk 2 above)
Comment (LazarusX) [reply src/container/acrManager.ts]: Another option is to simply throw a 404 error and https://github.com/Microsoft/vscode-azure-iot-edge/pull/412/files#diff-36614ddf00470008048ba066990c9c6fR145 will handle it.

File: src/container/acrManager.ts [hunk 3]
```diff
@@ -121,6 +122,12 @@ export class AcrManager {
 
             const repoItems: vscode.QuickPickItem[] = [];
             const repos = JSON.parse(catalogResponse).repositories;
+
+            if (!repos) {
+                isEmptyRegistry = true;
+                throw new Error("There is no repository in the registry.");
```
Comment (formulahendry) [inline src/container/acrManager.ts]: `"There is no repository in the registry."`  is this used?

Comment (SLdragon) [reply src/container/acrManager.ts]: yes, it is used in the error message

**Pull request merged**